src/tokens/tokens.service.ts

```diff
@@ -87,33 +85,22 @@ export class TokensService {
     this.factoryAddress = factoryAddress.toLowerCase();
     this.proxy.addConnectionListener(this);
     this.proxy.addEventListener(new TokenListener(this.mapper, this.blockchain));
-  }
-
-  async onConnect() {
     const wsUrl = new URL('/ws', this.baseUrl.replace('http', 'ws')).href;
-    const stream = await this.getStream(newContext());
-    this.proxy.configure(wsUrl, stream.name);
+    this.proxy.configure(wsUrl, this.topic);
   }
 
-  /**
-   * One-time initialization of event stream and base subscription.
-   */
-  async init(ctx: Context) {
-    this.stream = await this.getStream(ctx);
-    if (this.factoryAddress !== '') {
-      await this.createFactorySubscription(ctx, this.factoryAddress);
-    }
-  }
+  async onConnect() {}
```

Comment:
Seems like this can now be deleted, along with the call to `this.proxy.addConnectionListener(this)` in the constructor?

Reply:
Maybe eventstream-proxy doesn't need to track `connectListeners` at all...